src/mongocrypt.h.in

```diff
@@ -474,19 +441,30 @@ mongocrypt_setopt_append_csfle_search_path (mongocrypt_t *crypt,
  * @note This function will do no IO nor path validation. All validation will
  * occur during the call to @ref mongocrypt_init.
  *
- * @note If a CSFLE library path override is specified here, then no paths given
- * to @ref mongocrypt_setopt_append_csfle_search_path will be consulted when
- * opening the CSFLE library.
- *
- * @note If a path is provided via this API and @ref mongocrypt_init fails to
- * initialize a valid CSFLE library instance for the path specified, then
- * the initialization of mongocrypt_t will fail with an error.
+ * @note If a path is provided via this API, csfle is not disabled, and @ref
+ * mongocrypt_init fails to initialize a valid CSFLE library instance for the
+ * path specified, then the initialization of mongocrypt_t will fail with an
+ * error.
  */
 MONGOCRYPT_EXPORT
 void
 mongocrypt_setopt_set_csfle_lib_path_override (mongocrypt_t *crypt,
                                                const char *path);
 
+/**
+ * @brief Toggle disabling the csfle library.
+ *
+ * @param[in] crypt The @ref mongocrypt_t object to update
+ * @param[in] is_disabled Toggle whether csfle is disabled. If `true`, then
+ * mongocrypt_init will not search for csfle as part of initialization.
+ *
+ * @note This setting takes precedence over the
+ * @ref mongocrypt_setopt_set_csfle_lib_path_override.
```

Comment:
The proposed [spec change](https://github.com/mongodb/specifications/pull/1165/commits/54bd026d25dad755080aef4c6143f0f0400b2027#diff-de01d398114f63480af23798e09ffca599aa758efd04f15a9beec89b4566041bR994-R996) requires drivers fail when setting both `csfleDisabled` and `csflePathOverride`:

> If |opt-csfle-disabled| is |true| while either |opt-path-override| is defined
   or |opt-csfle-required| is |true|, this will cause initialization of
   encryption to unconditionally fail.

Consider having `mongocrypt_init` error if both `mongocrypt_setopt_set_csfle_lib_path_override` and `mongocrypt_setopt_set_csfle_disabled` are called. Drivers will still need to check if `csfleRequired` is set and `csfle` is not loaded. But checking in `mongocrypt_init` may simplify the set of checks drivers need to make.

Reply:
Added the check.